app/services/preserved_object_handler.rb

```diff
@@ -73,32 +74,41 @@ def check_existence
       results.add_result(AuditResults::INVALID_ARGUMENTS, errors.full_messages)
     elsif PreservedObject.exists?(druid: druid)
       Rails.logger.debug "check_existence #{druid} called"
-      transaction_ok = with_active_record_transaction_and_rescue do
-        raise_rollback_if_cm_po_version_mismatch
-
-        return results.report_results unless can_validate_current_comp_moab_status?
-
-        if incoming_version == comp_moab.version
-          set_status_as_seen_on_disk(true) unless comp_moab.status == 'ok'
-          results.add_result(AuditResults::VERSION_MATCHES, 'CompleteMoab')
-        elsif incoming_version > comp_moab.version
-          set_status_as_seen_on_disk(true) unless comp_moab.status == 'ok'
-          results.add_result(AuditResults::ACTUAL_VERS_GT_DB_OBJ, db_obj_name: 'CompleteMoab', db_obj_version: comp_moab.version)
-          if moab_validation_errors.empty?
-            comp_moab.upd_audstamps_version_size(ran_moab_validation?, incoming_version, incoming_size)
-            pres_object.current_version = incoming_version
-            pres_object.save!
-          else
-            update_status('invalid_moab')
+      # Make sure that moab exists on storage
```

Comment:
I feel like this method is getting way too big to understand what it's doing.  Can you refactor into smaller pieces?

Reply:
Does this help?